### ISSUE TITLE
AWS App Mesh adds support to connect services deployed in multiple AWS accounts into a shared mesh

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#12561]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS App Mesh now supports sharing a Mesh with multiple AWS accounts. 

### New or Affected Resource(s)

* [`aws_appmesh_mesh`](https://www.terraform.io/docs/providers/aws/r/appmesh_mesh.html)
* [`aws_appmesh_route`](https://www.terraform.io/docs/providers/aws/r/appmesh_route.html)
* [`aws_appmesh_virtual_node`](https://www.terraform.io/docs/providers/aws/r/appmesh_virtual_node.html)
* [`aws_appmesh_virtual_router`](https://www.terraform.io/docs/providers/aws/r/appmesh_virtual_router.html)
* [`aws_appmesh_virtual_service`](https://www.terraform.io/docs/providers/aws/r/appmesh_virtual_service.html)

### Potential Terraform Configuration

```hcl
# New attributes:
aws_appmesh_mesh.example.mesh_owner
aws_appmesh_mesh.example.resource_owner
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/03/aws-app-mesh-adds-support-to-connect-services-deployed-in-multiple-aws-accounts-into-shared-mesh/).

**Comment by ewbankkit** [comment]
Can start with cherry picking 0dfbf3b4508adafbf203b94a5fe2001a32a335a0 from https://github.com/terraform-providers/terraform-provider-aws/pull/11850 (and adding support into `aws_appmesh_mesh` which I forgot to do).

**Comment by bflad** [comment]
Support for this has been merged and will release with version 3.8.0 of the Terraform AWS Provider, later this week. Thanks to @ewbankkit for the implementation. 👍
